test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -1035,8 +1035,12 @@ private Object runExtensionMethod(ReflectiveInvocationContext<Method> invocation
                         cloneRequired = false;
                     } else if (TestInfo.class.isAssignableFrom(theclass)) {
                         TestInfo info = (TestInfo) arg;
+                        Method originalTestMethod = info.getTestMethod().get();
+                        Method newTestMethod = originalTestMethod.hashCode() == newMethod.hashCode()
```

Comment:
This is just a tiny optimization to avoid calling `determineTCCLExtensionMethod()` again when we have a "regular" test method (not before/after).
To avoid even more calls, we would need something like an `IdentiyHashMap` member in this class, but that felt like overkill for now.

Reply:
I'm not sure whether comparing hash codes is a good idea here. The current implementation of `Method.hashCode` in OpenJDK is only based on the class and method name, but not on the entire method signature (https://github.com/openjdk/jdk/blob/1fb798d320c708dfcbc0bb157511a2937fafb9e6/src/java.base/share/classes/java/lang/reflect/Method.java#L380), so if for some (stupid) reason someone writes a test class with the following two equally named methods:
```
@BeforeEach
public void method(TestInfo testInfo) {}

@Test
public void method() {}
```
the wrong method will be passed via `testInfo`.

I suggest comparing `originalTestMethod` with `invocationContext.getExecutable()` or `originalTestMethod.toString()` with `newMethod.toString()` instead. Both suggestions should work more reliable, though note that I did not test them.

Reply:
I was on the fence about this. Overloaded methods are not supported by Junit5, but I didn't think of the exact scenario you mentioned. I'd need to check what happens then.

If we want to be extra sure, we should probably drop that optimization entirely since Method.toString is not particularly cheap.

Reply:
I would just drop it and only adjust if it becomes a problem.

Reply:
Yeah, later today I plan to have a quick look at the IdentityHashMap solution and if that feels awkward I'll just go without any "caching".

Reply:
I pushed the `IdentityHashMap` approach. I like it, YMMV!

It avoids rescanning for methods that have already been determined before, also for non-TestInfo cases.
E.g. `@BeforeEach` methods are now looked up only once, not before _each_ actual test method execution.
Before that, a test with 10 actual test methods and one `@BeforeEach` and one `@AfterEach` method would trigger the reflective scan 30 times, with the cache it's only 12 times.

Since `QuarkusTestExtension` is recreated for each test class and that new cache is _not_ static, there is also no leak.

Reply:
PS, if in that example above those two `@BeforeEach` and `@AfterEach` methods had a `TestInfo` parameter, there would be 50 calls without a cache, but still only 12 with that cache.